### PR TITLE
feat: raise diagnostics on member instead of parent declaration

### PR DESCRIPTION
**Pull request by dsherret**

### CODE REVIEW

File: tests/specs/InternalTag.txt [hunk 1]
```diff
@@ -62,12 +62,6 @@ class Other
 
     @internal
 
-Defined in file:///mod.ts:4:1
-
-private type PrivateType = string
-
-  @internal
```
Comment (dsherret) [inline tests/specs/InternalTag.txt:69]: Bug.

File: src/util/swc.rs [hunk 2]
```diff
@@ -115,3 +115,8 @@ pub fn module_export_name_value(
     ModuleExportName::Str(str) => str.value.to_string(),
   }
 }
+
+/// If the jsdoc has an `@internal` or `@ignore` tag.
+pub fn has_ignorable_js_doc_tag(js_doc: &JsDoc) -> bool {
+  js_doc.tags.iter().any(|t| matches!(t, JsDocTag::Ignore) || matches!(t, JsDocTag::Unsupported { value } if value == "@internal" || value.starts_with("@internal ")))
+}
```
Comment (bartlomieju) [inline src/util/swc.rs:122]: Why isn't it line-wrapping here?

File: src/visibility.rs [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+      let decl_deps = symbol
+        .decls()
+        .iter()
+        .filter(|d| !d.has_overloads())
+        .flat_map(|decl| decl.deps().into_iter().map(move |dep| (decl, dep)))
+        .map(|(decl, dep)| (symbol, decl, dep))
+        .chain(
+          symbol
+            .members()
+            .iter()
```
Comment (bartlomieju) [inline src/visibility.rs:94]: This function is really complex, would it make sense to unit test it?

**Review by bartlomieju — APPROVED** [review]
LGTM, just a couple comments, but this is a great feature!

File: src/visibility.rs (hunk 3 above)
Comment (dsherret) [reply src/visibility.rs:94]: It's captured by the integration tests.

File: src/util/swc.rs (hunk 2 above)
Comment (dsherret) [reply src/util/swc.rs:122]: No idea. Rustfmt sometimes works in mysterious ways 🤷‍♂️

**Pull request merged**